crates/wasmi/src/module/mod.rs

```diff
@@ -132,8 +131,8 @@ impl Module {
     ///
     /// - If the `stream` cannot be decoded into a valid Wasm module.
     /// - If unsupported Wasm proposals are encountered.
-    pub fn new(engine: &Engine, stream: impl Read) -> Result<Self, Error> {
-        parse(engine, stream).map_err(Into::into)
+    pub fn new(engine: &Engine, stream: impl Read) -> Result<Self, ModuleError> {
+        parse(engine, stream)
```

Comment:
I'd merge if you revert this since it is not yet clear to me if we want to actually return `ModuleError` here. Generally speaking I want to refactor how errors are managed in `wasmi` since a long time now so unify error handling a bit.

Reply:
The user wants to know at which stage the error occurred and then deal with it. If it always returns Error, it needs to make another judgment.
Like linker.define returns LinkerError.